processor/transformprocessor/README.md

```diff
@@ -14,44 +14,131 @@
 
 | error_mode | description                                                                                                                                 |
 |------------|---------------------------------------------------------------------------------------------------------------------------------------------|
 | ignore     | The processor ignores errors returned by statements, logs the error, and continues on to the next statement.  This is the recommended mode. |
 | silent     | The processor ignores errors returned by statements, does not log the error, and continues on to the next statement.                        |
 | propagate  | The processor returns the error up the pipeline.  This will result in the payload being dropped from the collector.                         |
 
-If not specified, `propagate` will be used.
+### Flat Config
+
+> [!NOTE]
+> If you don't know how to write OTTL statements yet, first see [Writing OTTL](#writing-ottl).
+
+The flat configuration style allows you to configure OTTL statements as a list,
+without worrying about extra configurations.
+
+**This is the simplest way to configure the Transform Processor.** If you need global conditions or specific error modes see [Statement Group Config](#statement-group-config).
+
+Format:
+
+```yaml
+transform:
+  error_mode: ignore
+  <trace|metric|log>_statements:
+    - string
+    - string
+    - string
+```
+
+Example: 
+
+```yaml
+transform:
+  error_mode: ignore
+  trace_statements:
+    - keep_keys(span.attributes, ["service.name", "service.namespace", "cloud.region", "process.command_line"])
+    - replace_pattern(span.attributes["process.command_line"], "password\\=[^\\s]*(\\s?)", "password=***")
+    - limit(span.attributes, 100, [])
+    - truncate_all(span.attributes, 4096)
+  metric_statements:
+    - keep_keys(resource.attributes, ["host.name"])
+    - truncate_all(resource.attributes, 4096)
+    - set(metric.description, "Sum") where metric.type == "Sum"
+    - convert_sum_to_gauge() where metric.name == "system.processes.count"
+    - convert_gauge_to_sum("cumulative", false) where metric.name == "prometheus_metric"
+  log_statements:
+    - set(log.severity_text, "FAIL") where log.body == "request failed"
+    - replace_all_matches(log.attributes, "/user/*/list/*", "/user/{userId}/list/{listId}")
+    - replace_all_patterns(log.attributes, "value", "/account/\\d{4}", "/account/{accountId}")
+    - set(log.body, log.attributes["http.route"])
+```
+
+If you're interested in how OTTL parses these statements, see [Conetxt Inference](#context-inference).
+
+### Statement Group Config
+
+> [!NOTE]
+> If you don't know how to write OTTL statements yet, first see [Writing OTTL](#writing-ottl).
+
+For more complex use cases you may need to use Statement Groups to group related OTTL statements.
+
+```yaml
+  - error_mode: propagate
```

Comment:
Can you nest this in an `X_statements` object to make it more obvious how it fits into the overall config?

Reply:
updated